joplin/base/models/site_settings.py

```diff
@@ -41,9 +41,18 @@ class JanisBranchSettings(BaseSetting):
     ]
 
     # Convert branch name to a netlify URL
-    def branch_preview_url_base(self):
+    def url_base(self, janis_branch):
+        """
+        Returns base url depending on branch setting
+        """
+        # if input is a full URL, we don't need to construct one
         if self.preview_input == "url":
             return self.preview_janis_url
         else:
-            # Netlify site names are limited to 63 characters
-            return f"https://{('janis-' + self.preview_janis_branch)[:63]}.netlify.com"
+            try:
+                # Netlify site names are limited to 63 characters
+                return f"https://{('janis-' + getattr(self, janis_branch))[:63]}.netlify.com"
+            except AttributeError as e:
+                # TODO: better handiling, I think this defaults back to staging/prod as is
+                print("something missing", e)
```

Comment:
In the future, `except` clauses should handle errors or not exist at all. Showing a message as vague as "something missing" or "something wrong with flags maybe" is unhelpful or worse, passing breaking errors that should be addressed. 

It's a very good idea to handle potential errors here. But printing a statement and passing is not error handling. If error handling is a feature requirement that you don't want to tackle in this same PR, make a github issue to address it.

Reply:
we're still printing the error here, not sure if print is the best way to handle logging though, probably a bigger thing to think about

Reply:
for part of https://github.com/cityofaustin/techstack/issues/3185 I was going to see what it would take to convert these statments to use djangos logging. That way they can be controlled based on env settings and various verbosity stuff